File: Ruby/Bubble_Sort.rb
def is_sorted?(arr)
    len = arr.length

    0.upto(len-2) do |x|
        if arr[x] > arr[x+1]
            return false
        end
    end

    return true
end


def bubble_sort(arr)
    len = arr.length
    done = false

    while !done
        done = true
        0.upto(len-2) do |x|
            if arr[x] > arr[x+1]
                arr[x], arr[x+1] = arr[x+1], arr[x]
                done = false
            end
        end
        len -= 1
    end

    return arr
end


if __FILE__ == $0
    arr = 10.times.map { |x| (Random.rand() * 100).to_i + x }
    puts "Array #{arr}. Sorted? #{is_sorted?(arr)}"
    arr = bubble_sort(arr)
    puts "Array #{arr}. Sorted? #{is_sorted?(arr)}"
end

File: JavaScript/Bubble-Sort.js
function bubble_sort(arr) {
    for(var j=arr.length; j>=0; --j) {
        for(var i=0; i<j; ++i) {
            if(arr[i] > arr[i+1]) {
                var tmp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = tmp;
            }
        }
    }

    return arr;
}


var arr = [6, 4, 2, 1, 3, 5];
console.log(arr);
arr = bubble_sort(arr);
console.log(arr)
